### PR TITLE
DOC: Add example for Top-K with Others.

### PR DESCRIPTION
**Pull request by manzt**
Implementation of Top-K with Others [Example](https://vega.github.io/vega-lite/examples/window_top_k_others.html) from Vega-Lite website. Uses aggregate, window, and calculate transformations.

### CODE REVIEW

File: altair/examples/top_k_with_others.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+----------------------
+This example shows how to use aggregate, window, and calculate transfromations
+to display the top-k directors by average worldwide gross while grouping the 
+remaining directors as 'All Others'.
+"""
+# category: case studies
+import altair as alt
+from vega_datasets import data
+
+source = data.movies()
```
Comment (jakevdp) [inline altair/examples/top_k_with_others.py]: Please use `data.movies.url` (see https://github.com/altair-viz/altair/blob/master/CONTRIBUTING.md#steps-to-contribute for more info)

File: altair/examples/top_k_with_others.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        "ranked_director:N",
+        sort=alt.Sort(op="mean", field="aggregate_gross", order="descending"),
+        title=None,
+    ),
+).transform_aggregate(
+    aggregate=[
+        alt.AggregatedFieldDef(
+            **{"as": "aggregate_gross", "op": "mean", "field": "Worldwide_Gross"}
+        )
+    ],
```
Comment (jakevdp) [inline altair/examples/top_k_with_others.py]: Shorter:
```
aggregate_gross='mean(Worldwide_Gross)',
```

File: altair/examples/top_k_with_others.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    groupby=["Director"],
+).transform_window(
+    window=[alt.WindowFieldDef(**{"as": "rank", "op": "row_number"})],
```
Comment (jakevdp) [inline altair/examples/top_k_with_others.py]: Shorter: `rank='row_number()',`

File: altair/examples/top_k_with_others.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    sort=[alt.SortField("aggregate_gross", order="descending")],
+).transform_calculate(
+    as_="ranked_director", calculate="datum.rank < 10 ? datum.Director : 'All Others'"
```
Comment (jakevdp) [inline altair/examples/top_k_with_others.py]: shorter: `ranked_director="datum.rank < 10 ? datum.Director : 'All Others'"`

**Comment by manzt** [comment]
Much better with the higher-level syntax -- Thanks!

**Comment by jakevdp** [comment]
Looks great, thanks!

**Pull request merged**